File: src (cpp)/animal/reptilia/chameleon.cpp
#include "chameleon.h"

Chameleon::Chameleon(int _x, int _y, int _weight) : def_ID(10),
                                                    def_ratio_meat(20),
                                                    def_ratio_plant(50) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Chameleon::Interact() {
  std::cout << "The chameleon can be barely seen" << std::endl;
}

File: src/virtualzoo/animal/reptilia/Chameleon.java
package virtualzoo.animal.reptilia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Omnivora;

/**
 * Kelas Chameleon mendefinisikan atribut untuk chameleon.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Chameleon extends Reptilia {

  /**
   * Constructor
   *
   * Menciptakan chameleon dengan posisi (locX,locY) dan berat animalWeight.
   *
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat chameleon.
   */
  public Chameleon(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Omnivora(animalWeight, 20, 50);
    ID = 12;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan chameleon
   */
  public void interact() {
    System.out.println("The chameleon can be barely seen");
  }
}
